### ISSUE TITLE
Aspire container for RabbitMq not ready, crashing other Aspire managed apps using Wolverine/RabbitMq

### ISSUE THREAD

**Issue by yarseyah** [JasperFx/wolverine#752]
**Describe the bug**
I've been trying to use Aspire (Preview 3) to implement my architecture, using RabbitMq via Wolverine to get things communicating appropriately, I'm aware of the open #635 that seems to be working out how to use the Aspire client services, however in my case I'm purely using host to provide the necessary RabbitMq.  Within the Aspire launched applications (web-server, API, consumers etc) I want to use pure RabbitMq via Wolverine, e.g. not the `IConnection` etc that the Aspire client libraries add.  However, I'm having startup issues with the client crashing immediately.

**To Reproduce**
Steps to reproduce the behaviour:

1. Add to the Aspire host:
```csharp
var messageQueue = builder.AddRabbitMQ("command-queue");
```

2. Add to an Aspire invoked application
```csharp
builder.Host.UseWolverine(options => {
  var connectionString = builder.Configuration.GetConnectionString("command-queue")
                           ?? throw new InvalidOperationException();
  options.UseRabbitMq(new Uri(connectionString))
         .UseSenderConnectionOnly();    
});
```
3. Starting of client crashes with the following exception:

```
024-02-23T17:58:52.4817620 fail: Wolverine.Runtime.WolverineRuntime[0]
      Failed to start the Wolverine messaging
      RabbitMQ.Client.Exceptions.BrokerUnreachableException: None of the specified endpoints were reachable
       ---> System.IO.IOException: connection.start was never received, likely due to a network timeout
         at RabbitMQ.Client.Framing.Impl.Connection.StartAndTune()
         at RabbitMQ.Client.Framing.Impl.Connection.Open(Boolean insist)
         at RabbitMQ.Client.Framing.Impl.Connection..ctor(IConnectionFactory factory, Boolean insist, IFrameHandler frameHandler, String clientProvidedName)
         at RabbitMQ.Client.Framing.Impl.Connection..ctor(IConnectionFactory factory, Boolean insist, IFrameHandler frameHandler, ArrayPool`1 memoryPool, String clientProvidedName)
         at RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(IFrameHandler fh)
         at RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(IEndpointResolver endp
oints)
         at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName)
         --- End of inner exception stack trace ---
         at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName)
         at RabbitMQ.Client.ConnectionFactory.CreateConnection(String clientProvidedName)
         at RabbitMQ.Client.ConnectionFactory.CreateConnection()
         at Wolverine.RabbitMQ.Internal.RabbitMqTransport.BuildConnection()
         at Wolverine.RabbitMQ.Internal.RabbitMqTransport.ConnectAsync(IWolverineRuntime runtime)
         at Wolverine.Transports.BrokerTransport`1.InitializeAsync(IWolverineRuntime runtime)
         at Wolverine.Runtime.WolverineRuntime.startMessagingTransportsAsync()
         at Wolverine.Runtime.WolverineRuntime.StartAsync(CancellationToken cancellationToken)

```

This can be overcome by something hacky such as the following at the end of the program.cs for a minimal API :

```csharp
// Map routes

Console.WriteLine("Waiting for 5 seconds before starting the application");
await Task.Delay(5000);
Console.WriteLine("Starting the application after 5 seconds to allow RabbitMQ to start");

app.Run();
}
```

**Expected behaviour**
Initialisation code waits for the RabbitMq container to start.  (configurable timeout or resilience pattern?).

**Comment by jeremydmiller** [comment]
Wolverine doesn't yet support Aspire, here's where we're tracking that work: https://github.com/JasperFx/wolverine/issues/635

And no, I don't have a perfect idea of when that will be done, but it's on the radar.

**Comment by seralaci** [comment]
The [WaitFor()](https://learn.microsoft.com/en-us/dotnet/aspire/whats-new/dotnet-aspire-9-release-candidate-1?#waiting-for-dependencies) coming in 9.0 and will solve the originally raised issue, but until then you can use [David Fowler's prototype](https://github.com/davidfowl/WaitForDependenciesAspire/).

I have a [complete, working solution](https://github.com/seralaci/practical-wolverine/tree/main/projects/PingPongWithRabbitMq) that extends the PingPongWithRabbitMq example with (among other things) the Aspire integration.

**Comment by dombrovsky** [comment]
This looks less like an Aspire‑specific problem and more like Wolverine’s general behavior of trying to establish the RabbitMQ connection in IHostedService.StartAsync, which causes the host to fail/startup‑crash when the broker is unavailable instead of starting and reporting unhealthy.

To track this as a broader behavior (independent of Aspire wiring), I opened a separate issue https://github.com/JasperFx/wolverine/issues/1987
